sdk/data/aztables/README.md

```diff
@@ -1,151 +1,342 @@
+
+```bash
+az storage account show-connection-string -g MyResourceGroup -n MyStorageAccount
+```
+
+```golang
+connStr := "DefaultEndpointsProtocol=https;AccountName=<myAccountName>;AccountKey=<myAccountKey>;EndpointSuffix=core.windows.net"
+serviceClient, err := aztables.NewServiceClientFromConnectionString(connStr, nil)
+```
+
+##### Creating the client from a SAS token
+To use a [shared access signature (SAS) token][azure_sas_token], provide the token as a string. If your account URL includes the SAS token, omit the credential parameter. You can generate a SAS token from the Azure Portal under [Shared access signature](https://docs.microsoft.com/rest/api/storageservices/create-service-sas) or use the `ServiceClient.GetAccountSASToken` or `Client.GetTableSASToken()` methods.
+
+```golang
+cred, err := aztables.NewSharedKeyCredential("myAccountName", "myAccountKey")
+handle(err)
+service, err := aztables.NewServiceClient("https://<myAccountName>.table.core.windows.net", cred, nil)
+
+resources := aztables.AccountSASResourceTypes{Service: true}
+permission := aztables.AccountSASPermissions{Read: true}
+start := time.Date(2021, time.August, 21, 1, 1, 0, 0, time.UTC)
+expiry := time.Date(2022, time.August, 21, 1, 1, 0, 0, time.UTC)
+sasUrl, err := service.GetAccountSASToken(resources, permission, start, expiry)
+handle(err)
+
+sasService, err := aztables.NewServiceClient(sasUrl, azcore.AnonymousCredential(), nil)
+handle(err)
+```
 
-#### Creating a Cosmos DB
-
-Create a Cosmos DB account `MyCosmosDBDatabaseAccount` in resource group `MyResourceGroup`
-in the subscription `MySubscription` and a table named `MyTableName` in the account.
-
-
-### Authenticate the Client
-
-Learn more about options for authentication _(including Connection Strings, Shared Key, and Shared Key Signatures)_ in our samples.
 
 ## Key concepts
+Common uses of the table service include:
+* Storing TBs of structured data capable of serving web scale applications
+* Storing datasets that do not require complex joins, foreign keys, or stored procedures and can be de-normalized for fast access
+* Quickly querying data using a clustered index
+* Accessing data using the OData protocol filter expressions
+
+The following components make up the Azure Tables Service:
+* The account
+* A table within the account, which contains a set of entities
+* An entity within a table, as a dictionary
+
+The Azure Tables client library for Go allows you to interact with each of these components through the
+use of a dedicated client object.
+
+### Clients
+Two different clients are provided to interact with the various components of the Table Service:
+1. **`ServiceClient`** -
+    * Get and set account settings
+    * Query, create, and delete tables within the account.
+    * Get a `Client` to access a specific table using the `NewClient` method.
+2. **`Client`** -
```

Comment:
I'm guessing we think this is the one that people will use the most - should it be first in the list?